tutorials/polkadot-sdk/build-a-blockchain/add-trusted-nodes.md

```diff
@@ -0,0 +1,458 @@
+
+3. Use the generated secret phrase to derive keys using the Ed25519 signature scheme. Enter the same password you used in the previous step:
+
+    ```bash
+    ./target/release/solochain-template-node key inspect --password-interactive --scheme Ed25519 "INSERT_SECRET_PHRASE"
+    ```
+
+    !!! note
+        Replace `INSERT_SECRET_PHRASE` with the secret phrase generated in step 2.
+
+    The command will output information about the generated keys similar to the following:
+
+    ```plain
+    Secret phrase:       digital width rely long insect blind usual name oyster easy steak spend
+      Network ID:        substrate
+      Secret seed:       0xc52405d0b45dd856cbf1371f3b33fbde20cb76bf6ee440d12ea15f7ed17cca0a
+      Public key (hex):  0xc9c2cd111f98f2bf78bab6787449fc007dd7f2a5d02f099919f7fb50ade97dd6
+      Account ID:        0xc9c2cd111f98f2bf78bab6787449fc007dd7f2a5d02f099919f7fb50ade97dd6
+      Public key (SS58): 5GdFMFbXy24uz8mFZroFUgdBkY2pq6igBNGAq9tsBfEZRSzP
+      SS58 Address:      5GdFMFbXy24uz8mFZroFUgdBkY2pq6igBNGAq9tsBfEZRSzP
+    ```
+
+    The Ed25519 key you've generated is crucial for block finalization using the `grandpa` consensus algorithm. The Ed25519 public key for the account is `5GdFMFbXy24uz8mFZroFUgdBkY2pq6igBNGAq9tsBfEZRSzP`.
+
+### Generate a Second Set of Keys
+
+In this tutorial, the private network will consist of two nodes, meaning you'll need two distinct sets of keys. You have several options for generating these second set of keys:
+
+- Use the keys from one of the predefined accounts
+- Follow the steps from the previous section, but use a different identity on your local machine to create a new key pair
+- Derive a child key pair to simulate a second identity on your local machine
+
+For the purpose of this tutorial, the second set of keys will be:
+
+- Sr25519 (for Aura) - `5Df9bvnbqKNR8S1W2Uj5XSpJCKUomyymwCGf6WHKyoo3GDev`
+- Ed25519 (for Grandpa)  `5DJRQQWEaJart5yQnA6gnKLYKHLdpX6V4vHgzAYfNPT2NNuW`
+
+## Create a Custom Chain Specification
+
+After generating key pairs for your blockchain, the next step is creating a custom chain specification. This specification will be shared with trusted validators participating in your network.
+
+To enable others to participate in your blockchain, ensure that each participant generates their own key pair. Once you collect the keys from all network participants, you can create a custom chain specification to replace the default local one.
+
+In this tutorial, you'll modify the local chain specification to create a custom version for a two-node network. The same process can be used to add more nodes if you have the necessary keys.
+
+### Add Validators
+
+When creating your custom chain specification, you'll need to add validators by modifying two key sections:
```

Comment:
At least for me, it's unclear where I need to update/modify the plain chain spec. Maybe worth adding a reference to a custom chainspec or adding more steps here to provide more info

Reply:
In the Add Validators section, you don't have to add anything in the chain spec; it is more like an explanation. I agree that it is quite confusing

Reply:
Confirmed with Erin, please include the commands in terminal outputs.